eng/scripts/Update-CliVersion.ps1

```diff
@@ -0,0 +1,49 @@
+param(
+    [string] $NewVersion
+)
+
+$CLI_VERSION_FILE = "$PSScriptRoot../../../cli/version.txt"
+. "$PSScriptRoot../../common/scripts/common.ps1"
+
+Set-StrictMode -Version 3
+
+function getVersion {
+    $versionString = Get-Content $CLI_VERSION_FILE
+    return [AzureEngSemanticVersion]::new($versionString)
+}
+
+$version = $NewVersion
+$unreleased = $false
+$replaceLatestEntryTitle = $true
+
+if (!$version) {
+    # Increment after release
+    $version = getVersion
+
+    if ($version.IsPrerelease) {
+        if ($version.HasValidPrereleaseLabel()) {
+            # 0.1.0-beta.1 -> 0.1.0-beta.2
+            $version.PrereleaseNumber++
+        } else {
+            # 0.1.0 -> 0.2.0-beta.1
```

Comment:
Is this example correct?  I'm trying to understand how `0.1.0` would ever be considered pre-release and we'd go down this path.  I think maybe the example should be `0.1.0-badPrereleaseLabel.1 -> 0.2.0-beta.1`?

Reply:
From: https://semver.org/#spec-item-4 

> Major version zero (0.y.z) is for initial development. Anything MAY change at any time. The public API SHOULD NOT be considered stable.

The semver script that we have today lumps "initial development" in as a subset of "prerelease" though a good argument could probably be made that "initial development" is not pre-release. If we need to treat the two differently, we can do that. 

This does also handle the scenario you described.